src/katgpucbf/xbgpu/engine.py

```diff
@@ -719,7 +719,7 @@ def do_correlation() -> None:
                 self.correlation.first_batch = last_batch
 
         tx_item = await self._tx_free_item_queue.get()
-        await tx_item.async_wait_for_events()
+        tx_item.enqueue_wait_for_events(self._proc_command_queue)
```

Comment:
I'm pretty sure this is safe, for the trivial reason that this is in the initialisation code and hence the tx_item you get will be freshly minted without prior events anyway. To make a real difference you'd probably need to do the same thing at line 696. I think doing that will be free of race conditions, but will again cause a memory leak as `tx_item.events` will just keep growing (should be easy to check by printing it out each time).